// @ant-design/react-native/lib/locale-provider/es_ES.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _es_ES = require('../date-picker-view/locale/es_ES');

var _es_ES2 = _interopRequireDefault(_es_ES);

var _es_ES3 = require('../date-picker/locale/es_ES');

var _es_ES4 = _interopRequireDefault(_es_ES3);

var _es_ES5 = require('../input-item/locale/es_ES');

var _es_ES6 = _interopRequireDefault(_es_ES5);

var _es_ES7 = require('../list-view/locale/es_ES');

var _es_ES8 = _interopRequireDefault(_es_ES7);

var _es_ES9 = require('../modal/locale/es_ES');

var _es_ES10 = _interopRequireDefault(_es_ES9);

var _es_ES11 = require('../pagination/locale/es_ES');

var _es_ES12 = _interopRequireDefault(_es_ES11);

var _es_ES13 = require('../picker/locale/es_ES');

var _es_ES14 = _interopRequireDefault(_es_ES13);

var _es_ES15 = require('../search-bar/locale/es_ES');

var _es_ES16 = _interopRequireDefault(_es_ES15);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

exports['default'] = {
    locale: 'es',
    DatePicker: _es_ES4['default'],
    DatePickerView: _es_ES2['default'],
    InputItem: _es_ES6['default'],
    Modal: _es_ES10['default'],
    Pagination: _es_ES12['default'],
    Picker: _es_ES14['default'],
    SearchBar: _es_ES16['default'],
    ListView: _es_ES8['default']
};
module.exports = exports['default'];
