// @ant-design/react-native/lib/locale-provider/pt_BR.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _pt_BR = require('../date-picker-view/locale/pt_BR');

var _pt_BR2 = _interopRequireDefault(_pt_BR);

var _pt_BR3 = require('../date-picker/locale/pt_BR');

var _pt_BR4 = _interopRequireDefault(_pt_BR3);

var _pt_BR5 = require('../input-item/locale/pt_BR');

var _pt_BR6 = _interopRequireDefault(_pt_BR5);

var _pt_BR7 = require('../list-view/locale/pt_BR');

var _pt_BR8 = _interopRequireDefault(_pt_BR7);

var _pt_BR9 = require('../modal/locale/pt_BR');

var _pt_BR10 = _interopRequireDefault(_pt_BR9);

var _pt_BR11 = require('../pagination/locale/pt_BR');

var _pt_BR12 = _interopRequireDefault(_pt_BR11);

var _pt_BR13 = require('../picker/locale/pt_BR');

var _pt_BR14 = _interopRequireDefault(_pt_BR13);

var _pt_BR15 = require('../search-bar/locale/pt_BR');

var _pt_BR16 = _interopRequireDefault(_pt_BR15);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

exports['default'] = {
    locale: 'pt_BR',
    DatePicker: _pt_BR4['default'],
    DatePickerView: _pt_BR2['default'],
    InputItem: _pt_BR6['default'],
    Modal: _pt_BR10['default'],
    Pagination: _pt_BR12['default'],
    Picker: _pt_BR14['default'],
    SearchBar: _pt_BR16['default'],
    ListView: _pt_BR8['default']
};
module.exports = exports['default'];
